grpc/alteration_percent_test.go

```diff
@@ -0,0 +1,582 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package grpc_test
+
+import (
+	"fmt"
+
+	. "github.com/DataDog/chaos-controller/grpc"
+	pb "github.com/DataDog/chaos-controller/grpc/disruption_listener"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("get mapping from randomly generated Percent to Alteration based on AlterationToPercentAffected", func() {
+	var (
+		alterationSpecs []*pb.AlterationSpec
+		mapping         map[AlterationConfiguration]PercentAffected
+	)
+
+	Context("with three alterations which add up to less than 100", func() {
+		BeforeEach(func() {
+			alterationSpecs = []*pb.AlterationSpec{
+				{
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     int32(20),
+				},
+				{
+					ErrorToReturn:    "ALREADY_EXISTS",
+					OverrideToReturn: "",
+					QueryPercent:     int32(30),
+				},
+				{
+					ErrorToReturn:    "",
+					OverrideToReturn: "{}",
+					QueryPercent:     int32(40),
+				},
+			}
+
+			var err error
+			mapping, err = GetAlterationToPercentAffected(alterationSpecs)
+			Expect(err).To(BeNil())
+		})
+
+		It("should create a mapping with 3 elements", func() {
+			Expect(len(mapping)).To(Equal(3))
+		})
+
+		It("should create a mapping with correct configs", func() {
```

Comment:
Please let me know if I'm wrong, but it seems like the BeforeEach is going to be called before both of these `It`s? I think that's likely not your intention, and definitely is spending extra time than needed